File: src/main/java/com/flexpoker/web/commandsenders/InMemoryAsyncTableCommandSender.java
package com.flexpoker.web.commandsenders;

import com.flexpoker.framework.command.CommandReceiver;
import com.flexpoker.framework.command.CommandSender;
import com.flexpoker.table.command.framework.TableCommand;
import org.springframework.beans.factory.annotation.Qualifier;
import org.springframework.context.annotation.Lazy;
import org.springframework.stereotype.Component;

import javax.inject.Inject;

@Component
public class InMemoryAsyncTableCommandSender implements CommandSender<TableCommand> {

    private final CommandReceiver<TableCommand> tableCommandReceiver;

    @Lazy
    @Inject
    public InMemoryAsyncTableCommandSender(
            @Qualifier("tableCommandReceiver")
            CommandReceiver<TableCommand> tableCommandReceiver) {
        this.tableCommandReceiver = tableCommandReceiver;
    }

    @Override
    public void send(TableCommand command) {
        tableCommandReceiver.receive(command);
    }

}

File: src/main/kotlin/com/flexpoker/web/commandsenders/InMemoryAsyncTableCommandSender.kt
package com.flexpoker.web.commandsenders

import com.flexpoker.framework.command.CommandReceiver
import com.flexpoker.framework.command.CommandSender
import com.flexpoker.table.command.commands.TableCommand
import org.springframework.context.annotation.Lazy
import org.springframework.stereotype.Component
import javax.inject.Inject

@Component
class InMemoryAsyncTableCommandSender @Lazy @Inject constructor(
    private val tableCommandReceiver: CommandReceiver<TableCommand>
) : CommandSender<TableCommand> {

    override fun send(command: TableCommand) {
        tableCommandReceiver.receive(command)
    }

}